src/StreamsAPI_Lambda/Students/Main.java

```diff
@@ -0,0 +1,148 @@
+        namesByDepertament.forEach((departamentName, d) -> System.out.println(departamentName));
+
+
+        // count of students in each department
+        Map<String, Long> studentsByDepertament   = students.stream()
+                .collect(groupingBy(
+                        Students::getDepartamentName,//grupam dupa numele departamentuli
+                        counting()) //numaram membrii
+                );
+        System.out.println("\n Count department ");
+        studentsByDepertament.forEach((departament,count) -> System.out.println(departament + " : " +count));
+
+        //Find the list of students whose age is less than 30
+
+        List <Students> studentsUnder30 = students.stream()
+                .filter(students1 -> students1.getAge() < 30)//Find a list of students whose first name starts with 'A'
+                .collect(toList());
+        System.out.println("\n The list of students whose age is less than 30");
+        for (Students k : studentsUnder30) {
+            System.out.println(k);
+
+            //Find the list of students whose rank is between 50 and 100
+            List <Students> RankBetween50and100 = students.stream()
+                    .filter(students3 -> students3.getAverageGrade() >=50 && students3.getAverageGrade() <=100)//Find a list of students whose first name starts with 'A'
+                    .collect(toList());
+            System.out.println("\n The list of students whose rank is between 50 and 100");
+            for (Students l : RankBetween50and100) {
+                System.out.println(l);
+
+
+                //Find the average age of male and female students
+
+                Map<String, Double> averageAge   = students.stream()
+                        .collect(groupingBy(
+                                Students::getGender,//grupam dupa gen
+                                averagingInt(Students::getAge))); //calculam varsta;
+                System.out.println("\n The average age of male and female students  ");
+                studentsByDepertament.forEach((gender,averageAge1) -> System.out.println(gender + " : " + averageAge));
+
+
+                //Find the department which is having the maximum number of students
+
+                Map<String, Long> maxStudents   = students.stream()
+                        .collect(groupingBy(
+                                Students::getDepartamentName,//grupam dupa departament
+                                counting()
+                        ));//numara studentii
+
+                String departamentWithMaxStudents = maxStudents.entrySet().stream()
+                        .max(Map.Entry.comparingByValue()) // Găsește intrarea cu valoarea maximă (numărul maxim de studenți)
+                        .map(Map.Entry::getKey) // Extrage cheia (numele departamentului) din intrarea găsită
+                        .orElse("N/A"); // Dacă nu există nicio intrare, returnează "N/A" ca valoare implicită
```

Comment:
it looks like you could replace these 2 lines with `.get()` and it would still do the job